### PR TITLE
Add missing type hints for all flavors of LayoutLMv2 PyTorch models.

### PR DESCRIPTION
**Pull request by chainyo**
# What does this PR do?

I added type hints for all `LayoutLMv2` PyTorch classes as described in https://github.com/huggingface/transformers/issues/16059 .

@Rocketknight1

### CODE REVIEW

**Comment by Rocketknight1** [comment]
This also looks good, but same comment about the return types!

**Comment by ChainYo** [comment]
> This also looks good, but same comment about the return types!

Yes I fixed return types + I added all type hints for `LayoutLM` PyTorch

**Comment by Rocketknight1** [comment]
Hi, I'm seeing some tests failing on this one - I think you might have deleted a couple of arguments while adding the annotations!

File: src/transformers/models/layoutlm/modeling_layoutlm.py [hunk 1]
```diff
@@ -733,19 +734,17 @@ class PreTrainedModel
...
     def forward(
         self,
-        input_ids=None,
-        bbox=None,
-        attention_mask=None,
-        token_type_ids=None,
-        position_ids=None,
-        head_mask=None,
-        inputs_embeds=None,
-        encoder_hidden_states=None,
```
Comment (ChainYo) [inline src/transformers/models/layoutlm/modeling_layoutlm.py:743]: These arguments `encoder_hidden_states` and `encoder_attention_masks` are not used.

Should I fix tests too or it's out of the scope of this PR ?

File: src/transformers/models/layoutlm/modeling_layoutlm.py [hunk 2]
```diff
@@ -733,19 +734,17 @@ class PreTrainedModel
...
-        encoder_attention_mask=None,
```
Comment (ChainYo) [inline src/transformers/models/layoutlm/modeling_layoutlm.py:744]: This one is unused too

Comment (Rocketknight1) [reply src/transformers/models/layoutlm/modeling_layoutlm.py:744]: I think it's out of scope of this PR, but good catch nonetheless! I'm not sure why the arguments are there if they're not used, but you can raise an issue after this PR is merged if you like.

**Review by Rocketknight1 — APPROVED** [review]
Looks great, thank you!

**Pull request merged**